inst/scripts/make-zilionis-lung-data.Rmd

```diff
@@ -0,0 +1,187 @@
+Counts for endogenous genes are available from the Gene Expression Omnibus
+using the accession number [GSE127465](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE127465).
+We download and cache it using the `r Biocpkg("BiocFileCache")` package.
+
+```{r}
+library(BiocFileCache)
+bfc <- BiocFileCache(ask=FALSE)    
+tarball <- bfcrpath(bfc, 
+    "https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE127465&format=file")
+```
+
+We unpack it to a temporary directory.
+
+```{r}
+temp <- tempfile()
+untar(tarball, exdir=temp)
+```
+
+# Reading in human data
+
+We set up a function to load in each set of counts as a sparse matrix.
+
+```{r}
+library(Matrix)
+FUN <- function(X) {
+    input <- read.csv(X, sep="\t", stringsAsFactors=FALSE)
+    labels <- as.character(input$barcode)
+    input <- input[,2:ncol(input)]
+    input <- t(input)
+    list(mat=as(input, "dgCMatrix"), label=labels)
+}
+```
+
+We read in all the human datasets.
+
+```{r}
+hs.files <- c(
+    "GSM3635278_human_p1t1_raw_counts.tsv.gz",
+    "GSM3635279_human_p1t2_raw_counts.tsv.gz",
+    "GSM3635280_human_p1t3_raw_counts.tsv.gz",
+    "GSM3635281_human_p1t4_raw_counts.tsv.gz",
+    "GSM3635282_human_p1b1_raw_counts.tsv.gz",
+    "GSM3635283_human_p1b2_raw_counts.tsv.gz",
+    "GSM3635284_human_p1b3_raw_counts.tsv.gz",
+    "GSM3635285_human_p2t1_raw_counts.tsv.gz",
+    "GSM3635286_human_p2t2_raw_counts.tsv.gz",
+    "GSM3635287_human_p2b1_raw_counts.tsv.gz",
+    "GSM3635288_human_p3t1_raw_counts.tsv.gz",
+    "GSM3635289_human_p3t2_raw_counts.tsv.gz",
+    "GSM3635290_human_p3t3_raw_counts.tsv.gz",
+    "GSM3635291_human_p3b1_raw_counts.tsv.gz",
+    "GSM3635292_human_p4t1_raw_counts.tsv.gz",
+    "GSM3635293_human_p4t2_raw_counts.tsv.gz",
+    "GSM3635294_human_p4t3_raw_counts.tsv.gz",
+    "GSM3635295_human_p4b1_raw_counts.tsv.gz",
+    "GSM3635296_human_p5t1_raw_counts.tsv.gz",
+    "GSM3635297_human_p5t2_raw_counts.tsv.gz",
+    "GSM3635298_human_p6t1_raw_counts.tsv.gz",
+    "GSM3635299_human_p6t2_raw_counts.tsv.gz",
+    "GSM3635300_human_p6b1_raw_counts.tsv.gz",
+    "GSM3635301_human_p7t1_raw_counts.tsv.gz",
+    "GSM3635302_human_p7t2_raw_counts.tsv.gz",
+    "GSM3635303_human_p7b1_raw_counts.tsv.gz"
+)
+all.human <- lapply(file.path(temp, hs.files), FUN)
+sapply(all.human, function(x) dim(x$mat))
+```
+
+We verify that the gene order is the same, and combine the counts.
+
+```{r}
+stopifnot(length(unique(lapply(all.human, function(x) rownames(x$mat))))==1L)
+counts <- do.call(cbind, lapply(all.human, function(x) x$mat))
+dim(counts)
+```
+
+We do the same thing with the column metadata.
+
+```{r}
+samples <- sapply(stringi::stri_split_fixed(hs.files, pattern = "_"), function(x) x[3])
+barcode <- lapply(all.human, function(x) x$label)
+origin <- rep(samples, times = sapply(barcode, length))
+donor <- stringi::stri_match_first_regex(origin, "p[0-9]+")[,1]
+replicate <- stringi::stri_match_first_regex(origin, "[t|b][0-9]+")[,1]
```

Comment:
A simpler approach would be `sub("[bt].*", "", origin)` and also `sub("p.", "", origin)`.